### DISCUSSION TITLE
Undefined reference to temp_file

### DISCUSSION THREAD

**Discussion by zrd1234** [coin-or/CppAD#173] [Q&A]
Hello, after installing CppAD on arm architecture, I wrote a test file to test it. When I compiled it, I encountered the following error.
<img width="558" alt="1680470580041" src="https://user-images.githubusercontent.com/96851901/229418489-03015bc5-f6b7-439d-8e24-2a1244322607.png">

**Comment by bradbell** [comment]
How did you solve your problem ? It seems from the message above that you were missing -lcppad_lib on the link line ?

**Comment by zrd1234** [comment]
> How did you solve your problem ? It seems from the message above that you were missing -lcppad_lib on the link line ?
Just connect the library to the terminal prompt.

**Comment by ShyLvy** [comment]
Lower the Cppad version,I use the 20220000.0 and it works.

**Comment by bradbell** [comment]
Checking version 20220000
https://github.com/coin-or/CppAD/tree/stable/20220000/cppad_lib

The temp_file utility is not in cppad_lib
https://github.com/coin-or/CppAD/tree/stable/20220000/cppad_lib

I suspect that the following answer would solve your problem in the current version:
https://github.com/coin-or/CppAD/discussions/182

**Comment by bradbell** [comment]
On unix systems, perhaps this is a good way to solve this problem.

1. Let prefix be the cppad_prefix; see
   https://cppad.readthedocs.io/en/latest/cmake.html#cppad-prefix
2. Execute the command ``find prefix -name='*.pc'``
3. Let dir be one of the directories where ``cppad.pc`` is located
4. Execute the command ``export PKG_CONFIG_PATH=dir``
5. Execute the command ``pkg-config --libs --cflags cppad``

The output of the last command above will give you all the flags necessary to link cppad.
